UniswapV3Adapter/contracts/UniswapV3Adapter.sol

```diff
@@ -0,0 +1,99 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "./UniswapV3likeAdapter.sol";
+
+interface IUniV3Factory {
+    function feeAmountTickSpacing(uint24) external view returns (int24);
+
+    function getPool(
+        address,
+        address,
+        uint24
+    ) external view returns (address);
+}
+
+contract UniswapV3Adapter is UniswapV3likeAdapter {
+    using SafeERC20 for IERC20;
+
+    address immutable FACTORY;
+    mapping(uint24 => bool) public isFeeAmountEnabled;
+    uint24[] public feeAmounts;
+
+    constructor(
+        string memory _name,
+        uint256 _swapGasEstimate,
+        uint256 _quoterGasLimit,
+        address _quoter,
+        address _factory
+    ) UniswapV3likeAdapter(_name, _swapGasEstimate, _quoter, _quoterGasLimit) {
+        require(_factory != address(0));
+        addDefaultFeeAmounts();
+        FACTORY = _factory;
+    }
+
+    function addDefaultFeeAmounts() internal {
+        addFeeAmount(500);
+        addFeeAmount(3000);
+        addFeeAmount(10000);
+    }
+
+    function enableFeeAmounts(uint24[] calldata _amounts)
+        external
+        onlyMaintainer
+    {
+        for (uint256 i; i < _amounts.length; ++i) enableFeeAmount(_amounts[i]);
+    }
+
+    function enableFeeAmount(uint24 _fee) internal {
+        require(!isFeeAmountEnabled[_fee], "Fee already enabled");
+        if (IUniV3Factory(FACTORY).feeAmountTickSpacing(_fee) == 0)
+            revert("Factory doesn't support fee");
+        addFeeAmount(_fee);
+    }
+
+    function addFeeAmount(uint24 _fee) internal {
+        isFeeAmountEnabled[_fee] = true;
+        feeAmounts.push(_fee);
+    }
+
+    function getBestPool(address token0, address token1)
```

Comment:
Missing NatSpec
Add descriptions for the interface, events and functions
Refer https://docs.soliditylang.org/en/v0.8.17/natspec-format.htm